packages/nestjs/src/setup.ts

```diff
@@ -0,0 +1,156 @@
+      return next.handle();
+    }
+
+    if (context.getType() === 'http') {
+      const req = context.switchToHttp().getRequest();
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (req.route) {
+        // eslint-disable-next-line @sentry-internal/sdk/no-optional-chaining,@typescript-eslint/no-unsafe-member-access
+        getIsolationScope().setTransactionName(`${req.method?.toUpperCase() || 'GET'} ${req.route.path}`);
+      }
+    }
+
+    return next.handle();
+  }
+}
+Injectable()(SentryTracingInterceptor);
+
+/**
+ * Global filter to handle exceptions and report them to Sentry.
+ */
+export class SentryGlobalFilter extends BaseExceptionFilter {
+  /**
+   * Catches exceptions and reports them to Sentry unless they are expected errors.
+   */
+  public catch(exception: unknown, host: ArgumentsHost): void {
+    const status_code = (exception as { status?: number }).status;
+
+    // don't report expected errors
+    if (status_code !== undefined && status_code >= 400 && status_code < 500) {
+      return super.catch(exception, host);
+    }
+
+    captureException(exception);
+    return super.catch(exception, host);
+  }
+}
+Catch()(SentryGlobalFilter);
+
+/**
+ * Service to set up Sentry performance tracing for Nest.js applications.
+ */
+export class SentryIntegrationService implements OnModuleInit {
+  /**
+   * Initializes the Sentry integration service and registers span attributes.
+   */
+  public onModuleInit(): void {
+    // Sadly, NestInstrumentation has no requestHook, so we need to add the attributes here
+    // We register this hook in this method, because if we register it in the integration `setup`,
+    // it would always run even for users that are not even using Nest.js
+    const client = getClient();
+    if (client) {
+      client.on('spanStart', span => {
+        addNestSpanAttributes(span);
+      });
+    }
+  }
+}
+Injectable()(SentryIntegrationService);
+
+/**
+ * Set up a root module that can be injected in nest applications.
+ */
+export class SentryIntegrationModule {
+  /**
+   * Configures the module as the root module in a Nest.js application.
+   */
+  public static forRoot(): DynamicModule {
+    return {
+      module: SentryIntegrationModule,
+      providers: [
+        SentryIntegrationService,
+        {
+          provide: APP_FILTER,
+          useClass: SentryGlobalFilter,
+        },
+        {
+          provide: APP_INTERCEPTOR,
+          useClass: SentryTracingInterceptor,
+        },
+      ],
+      exports: [SentryIntegrationService],
+    };
+  }
+}
+Global()(SentryIntegrationModule);
+Module({
+  providers: [
+    SentryIntegrationService,
+    {
+      provide: APP_FILTER,
+      useClass: SentryGlobalFilter,
+    },
+    {
+      provide: APP_INTERCEPTOR,
+      useClass: SentryTracingInterceptor,
+    },
+  ],
+  exports: [SentryIntegrationService],
+})(SentryIntegrationModule);
+
+function addNestSpanAttributes(span: Span): void {
```

Comment:
l: Since this function is just used once I would just inline it, which would make it one less indirection. It's stylistic preference though so I will leave it up to you.

Reply:
I'll leave this as is, because I think it is a bit easier to understand what is going on with the function name as context